src/index.js

```diff
@@ -0,0 +1,253 @@
+const events = require('events');
+const debug = require('debug');
+const amqplib = require('amqplib');
+const assert = require('assert');
+
+var connectorCounter = 0;
+
+/**
+ * Build Pulse ConnectionString, from options on the form:
+ * {
+ *   username:          // Pulse username
+ *   password:          // Pulse password
+ *   hostname:          // Hostname to use (defaults to pulse.mozilla.org)
+ * }
+ */
+const buildPulseConnectionString = function({username, password, hostname}) {
+  assert(username, 'options.username password is required');
+  assert(password, 'options.password is required');
+
+  // Construct connection string
+  return [
+    'amqps://',         // Ensure that we're using SSL
+    username,
+    ':',
+    password,
```

Comment:
This isn't new with this new library, but should we be url-encoding username and password?

I generated a weird password with some sort of meaningful url character for tc-github the other day and it fell over when I used it.